e2e-identity/tests/utils/fmk.rs

```diff
@@ -35,6 +35,30 @@ use crate::utils::{
 pub(crate) static mut GOOGLE_SND: Option<std::sync::Mutex<std::sync::mpsc::Sender<String>>> = None;
 static mut GOOGLE_RECV: Option<std::sync::Mutex<std::sync::mpsc::Receiver<String>>> = None;
 
+fn keypair_to_pubkey(alg: JwsAlgorithm, keypair: &Pem) -> Pem {
+    match alg {
+        JwsAlgorithm::P256 => ES256KeyPair::from_pem(keypair)
+            .unwrap()
+            .public_key()
+            .to_pem()
+            .unwrap()
+            .into(),
+        JwsAlgorithm::P384 => ES384KeyPair::from_pem(keypair)
+            .unwrap()
+            .public_key()
+            .to_pem()
+            .unwrap()
+            .into(),
+        JwsAlgorithm::P521 => ES512KeyPair::from_pem(keypair)
+            .unwrap()
+            .public_key()
+            .to_pem()
+            .unwrap()
+            .into(),
```

Comment:
Is it possible to factor some of these calls out and call them on the match result? Or is it a distinct type after every call?

Reply:
Unfortunately, the latter (yeah, it bothered me as well, but the API is what it is...).